packages/react/src/ElementFactory.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from "react";
+import { MantineProvider } from "@mantine/core";
+import { BlockNoteTheme } from "./BlockNoteTheme";
+import Tippy, { TippyProps } from "@tippyjs/react";
+import { createRoot } from "react-dom/client";
+import { EditorElement, RequiredDynamicParams } from "@blocknote/core";
+
+export const ElementFactory = <
```

Comment:
Pretty nice that you've extracted this to a single element for all the different factories, that cleans up code quite a bit. I don't see any per-factory exceptions (you've been able to keep it generic), so that's really nice.

Please do at some comments to what ElementFactory is, because it's now a pretty important centerpiece of the library